### PR TITLE
[script.speedtester@matrix] 1.1.0+matrix.1

### PR DESCRIPTION
**Pull request by dagwieers**
### Add-on details:

- **General**
  - Add-on name: Speed Tester
  - Add-on ID: script.speedtester
  - Version number: 1.1.0+matrix.1
  - Kodi/repository version: matrix

- **Code location**
  - URL: https://github.com/add-ons/script.speedtester
  
This add-on tests your Internet bandwith and reports both download and upload speeds and latency from within Kodi.

Use this if you are experiencing streaming issues and you suspect your network to be the cause.

### Description of changes:

v1.1.0 (2020-09-26)
- Move add-on to Add-ons project
- Fix typos and structure of standard output

v1.0.1 (2019-08-06)
- Initial release
    

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

### CODE REVIEW

**Comment by dagwieers** [comment]
Because the original author (@Dr0idGuy) has not responded for the past 9 months (https://github.com/Dr0idGuy/Kodi-Addons/pull/1), I forked the project and moved it to the Add-ons org.

A large cleanup is needed as well, but this submission only fixes a few known issues.

**Comment by dagwieers** [comment]
@Dr0idGuy responded by updating his repository (https://github.com/Dr0idGuy/Kodi-Addons) and referring to the new one. So he is no longer MIA ;-)

I have invited him to become admin on this new repository as well.

**Comment by enen92** [comment]
@Dr0idGuy can you please comment wether you're fine with this? 
This PR won't be merged unless you're okay with it (main author and not inactive)

**Comment by dagwieers** [comment]
@enen92 Did you look at his repository: https://github.com/Dr0idGuy/Kodi-Addons ?

**Comment by enen92** [comment]
Nop, I did not sorry. So it was moved into "your" repo with his blessing. Should be fine then, I'll take care of the review.
Thanks for the heads up

File: script.speedtester/default.py [hunk 1]
```diff
@@ -0,0 +1,939 @@
...
+import socket
+import timeit
+import platform
+import threading
+
+ADDON = xbmcaddon.Addon()
+ADDONID = ADDON.getAddonInfo('id')
+ADDONNAME = ADDON.getAddonInfo('name')
+ADDONVERSION = ADDON.getAddonInfo('version')
+ART = xbmc.translatePath(os.path.join(ADDON.getAddonInfo('path'),'resources', 'skins', 'Default', 'media'))
```
Comment (enen92) [inline script.speedtester/default.py:69]: xbmcvfs please

File: script.speedtester/default.py [hunk 2]
```diff
@@ -0,0 +1,939 @@
...
+
+def getConfig():
+    request = \
+        build_request('http://www.speedtest.net/speedtest-config.php')
+    uh = catch_request(request)
+    if uh is False:
+        xbmc.log('Could not retrieve speedtest.net configuration: %s' % e,xbmc.LOGDEBUG)
+        sys.exit(1)
+    configxml = []
+    while 1:
```
Comment (enen92) [inline script.speedtester/default.py:215]: Can you write the loop properly with abortrequested checks?

File: script.speedtester/default.py [hunk 3]
```diff
@@ -0,0 +1,939 @@
...
+            self.button_close_glow = xbmcgui.ControlImage(880, 418, 300, 122, '', aspectRatio=0)
+            self.addControl(self.button_close_glow)
+            self.button_close_glow.setVisible(False)
+            self.button_close_glow.setImage(self.image_button_run_glow)
+            self.button_close_glow.setAnimations([
+                ('conditional',
+                'effect=fade start=0 time=1000 delay=2000 pulse=true condition=Control.IsVisible(%d)' % self.button_close_glow.getId())
+            ])
+
+            self.button_close = xbmcgui.ControlButton(99999, 99999, 300, 122, "[B]Close[/B]",focusTexture=self.image_button_run,noFocusTexture=self.image_button_run, alignment=2 | 4,textColor='0xFF000000', focusedColor='0xFF000000',shadowColor='0xFFCCCCCC', disabledColor='0xFF000000')
```
Comment (enen92) [inline script.speedtester/default.py:443]: Hardcoded close

File: script.speedtester/default.py [hunk 4]
```diff
@@ -0,0 +1,939 @@
...
+        for size in sizes:
+            for i in range(0, 4):
+                urls.append('%s/random%sx%s.jpg' %
+                                    (os.path.dirname(best['url']), size, size))
+        self.imgGauge.setVisible(True)
+        xbmc.Monitor().waitForAbort(1)
+        self.configGauge(0)
+        self.imgGauge_arrow.setVisible(True)
+
+        startST.append('Testing download speed')
```
Comment (enen92) [inline script.speedtester/default.py:872]: Hardcoded non localized value

File: script.speedtester/default.py [hunk 5]
```diff
@@ -0,0 +1,939 @@
...
+        self.update_textbox(startST)
+        self.dl_textbox.setLabel("%.2f" % float((dlspeed / 1000 / 1000) * units[1]))
+
+        sizesizes = [int(.25 * 1000 * 1000), int(.5 * 1000 * 1000)]
+        sizes = []
+        for size in sizesizes:
+            for i in range(0, 25):
+                sizes.append(size)
+
+        startST.append('Testing upload speed')
```
Comment (enen92) [inline script.speedtester/default.py:885]: Same

File: script.speedtester/default.py [hunk 6]
```diff
@@ -0,0 +1,939 @@
...
+        self.update_textbox(startST)
+        ulspeed = self.uploadSpeed(best['url'], sizes, simple)
+        startST.append('Upload: %0.2f M%s/s' % ((ulspeed / 1000 / 1000) * units[1], units[0]))
+        self.update_textbox(startST)
+        self.ul_textbox.setLabel("%.2f" % float((ulspeed / 1000 / 1000) * units[1]))
+        self.configGauge(0, (ulspeed / 1000 / 1000) * 8, time=3000)
+        xbmc.Monitor().waitForAbort(2)
+
+        if share:
```
Comment (enen92) [inline script.speedtester/default.py:894]: Same below

**Review by enen92 — CHANGES_REQUESTED** [review]
See comments

**Comment by dagwieers** [comment]
So the plan was to first do one release which fixes a few things but with minimal changes, and then fix all the sanity checks and constructions. I don't mind fixing all these, but this will become v1.1.1 as v1.1.0 has already been released.

File: script.speedtester/default.py (hunk 1 above)
Comment (dagwieers) [reply script.speedtester/default.py:69]: Sure.

File: script.speedtester/default.py (hunk 2 above)
Comment (dagwieers) [reply script.speedtester/default.py:215]: This should be simplified a lot. We have code in other projects that do this more elegant :-)

File: script.speedtester/default.py (hunk 3 above)
Comment (dagwieers) [reply script.speedtester/default.py:443]: Not sure what you mean here. Can you elaborate?

File: script.speedtester/default.py (hunk 4 above)
Comment (dagwieers) [reply script.speedtester/default.py:872]: Yes, they need to go, but do we have to do that now?
It is English-only now anyway, and the generated image will also be English.

File: script.speedtester/default.py (hunk 6 above)
Comment (dagwieers) [reply script.speedtester/default.py:894]: Same what ?

Comment (dagwieers) [reply script.speedtester/default.py:894]: Same what?

File: script.speedtester/default.py (hunk 3 above)
Comment (enen92) [reply script.speedtester/default.py:443]: "Close" has to be localized

File: script.speedtester/default.py (hunk 6 above)
Comment (enen92) [reply script.speedtester/default.py:894]: hardcoded non-localized values

Comment (dagwieers) [reply script.speedtester/default.py:894]: I don't see any below, only log messages.

**Comment by Dr0idGuy** [comment]
> @Dr0idGuy can you please comment wether you're fine with this?
> This PR won't be merged unless you're okay with it (main author and not inactive)

@enen92 Im ok with this, I haven't been coding (on the side) much but will try to help @dagwieers to get this script up to date.

File: script.speedtester/default.py (hunk 6 above)
Comment (enen92) [reply script.speedtester/default.py:894]: disregard, it is just query arguments for the http request

**Comment by enen92** [comment]
@dagwieers I'm normally fine if updates to the addon come at a later time to fix minor things such as localized strings. In this case this is actually a new submission (addon is new in matrix), so I'd appreciate if you could just fix those minors before this gets merged (it's not much work actually). Code cleanup and improvements (the addon needs it) can well come afterwards

**Comment by dagwieers** [comment]
@enen92 Localized strings is not a minor change. I did a complete cleanup already, only the localization needs to happen.

The thing is, I want a lot more things to change and localizing it now makes no sense unless everything else has happened. So yes, your minor requirement becomes a major update. https://github.com/add-ons/script.speedtester

**Comment by enen92** [comment]
@dagwieers whatever...as long as you keep it in mind as a future requirement. Please change the xbmc -> xbmcvfs for this submission as we plan to drop translatePath in v20 (and the method is already in the python api for v19).

**Pull request closed without merging**